### PR TITLE
Support searching for multiple error messages when retriggering a job 

### PR DESCRIPTION
**Pull request by andromaqui**
# Description

This pr adds changes which enable specifying multiple error messages to search for when attempting to retrigger a github run.

## Related
https://github.com/camunda/team-infrastructure/issues/572

## Tests

Test 1: The workflow wont get retriggered if none of the error messages have been encountered
[Commit](https://github.com/camunda/infra-global-github-actions/pull/161/commits/dc5140c4e3724abba6cafabdf2387b1d8cf853f9)
[Test workflow invoking the composite action ](https://github.com/camunda/infra-global-github-actions/blob/dc5140c4e3724abba6cafabdf2387b1d8cf853f9/.github/workflows/test-worflow.yml)
[Workflow run](https://github.com/camunda/infra-global-github-actions/actions/runs/8067095805)

Test 2: One of the error messages is encountered in the logs
[Commit](https://github.com/camunda/infra-global-github-actions/pull/161/commits/85d2b4f7733adcfda27a6be5442bee995acae81d) 
[Test workflow invoking the composite action](https://github.com/camunda/infra-global-github-actions/pull/161/commits/85d2b4f7733adcfda27a6be5442bee995acae81d)
Workflow runs [1](https://github.com/camunda/infra-global-github-actions/actions/runs/8067471724) and [2](https://github.com/camunda/infra-global-github-actions/actions/runs/8067463033) 
Screenshot 2024-02-27 at 16 46 17

Test 3: Single line error messages are still working
[Commit](https://github.com/camunda/infra-global-github-actions/pull/161/commits/c3258bf4f307203d79982d8aacc62415b7a0c534)
[Test workflow invoking the composite action](https://github.com/camunda/infra-global-github-actions/blob/01a363a3121ef9e7c7543d14c98a923f62851b7f/.github/workflows/test-worflow.yml)
Workflow runs [1](https://github.com/camunda/infra-global-github-actions/actions/runs/8067676954) and [2](https://github.com/camunda/infra-global-github-actions/actions/runs/8067686781)
Screenshot 2024-02-27 at 17 02 34

Test 4: Triggering the workflow through the webmodeler repo with multiple error messages retriggers the action if it fails
PR https://github.com/camunda/web-modeler/pull/8087
Commit https://github.com/camunda/web-modeler/pull/8087/commits/ad8e1be1c12f0f98a70c71f1bf01c7c0225cad68
[Test workflow invoking the composite action ](https://github.com/camunda/web-modeler/actions/runs/8068810141/workflow?pr=8087)
Workflow run [attempt 1](https://github.com/camunda/web-modeler/actions/runs/8068810141/attempts/1?pr=8087) and [attempt 2](https://github.com/camunda/web-modeler/actions/runs/8068810141/job/22042480397?pr=8087)

### CODE REVIEW

File: rerun-failed-run/action.yml [hunk 1]
```diff
@@ -51,4 +51,25 @@ runs:
...
+        Process completed, with exit code 99
+        EOF
+        )
+        json_array=()
+        counter=1
+        # Read each line of the multiline string
+        IFS=$'\n'
+        for line in $multiline_variable
+        do
+           json_array+="\"error$counter\": \"$line\", "
```
Comment (maxdanilov) [inline rerun-failed-run/action.yml:66]: I don't really see why produce a map here (`{"error1": "...", "error2": ...}`) if it's much easier to make a flat list of strings (`["...", "..."]`), like this:

```bash
multiline_variable=$(cat << EOF
Process completed with exit code 88
Process completed, with exit code 99
EOF
)
result="["

for line in $multiline_variable
do
  result+="\"$line\","
done

result=$(echo $result | sed '$ s/,$//g')
result+="]"

echo $result
```

Comment (andromaqui) [reply rerun-failed-run/action.yml:66]: yeah this actually makes much more sense, will address this

**Review by clementnero — CHANGES_REQUESTED** [review]
Thx for all the detailed explanations 🙇 
I have added some comments directly in the code.

File: rerun-failed-run/action.yml [hunk 2]
```diff
@@ -51,4 +51,14 @@ runs:
...
+        error_messages="${{ inputs.error-messages }}"
+        result="["
+        IFS=$'\n'
+        for line in $error_messages
+        do
+          result+="\"$line\","
+        done
+        result=$(echo $result | sed '$ s/,$//g')
+        result+="]"
+        echo "result: $result"
```
[the remark this answers is not in the document]
Comment (andromaqui) [reply rerun-failed-run/action.yml]: true much more elegant

File: .github/workflows/README.md [hunk 3]
```diff
@@ -62,5 +62,6 @@ jobs:
         shell: bash
         run: |
           echo ${{ steps.github-token.outputs.token }} | gh auth login --with-token
-          gh workflow run rerun-failed-run.yml -R camunda/infra-global-github-actions --ref=main -F repository=${{ github.repository }} -F error_message="The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled." -F run_id=${{ github.run_id }}
+          error_messages="[\"First error message\", \"Second error message\"]"
+          gh workflow run rerun-failed-run.yml -R camunda/infra-global-github-actions --ref=main -F repository=${{ github.repository }} -F error_messages="The runner has received a shutdown signal. This can happen when the runner service is stopped, or a manually started runner is canceled." -F run_id=${{ github.run_id }}
```
[the remark this answers is not in the document]
Comment (andromaqui) [reply .github/workflows/README.md:66]: yes !

File: .github/workflows/rerun-failed-run.yml [hunk 4]
```diff
@@ -46,16 +46,22 @@ jobs:
...
+              echo "error_count: $error_count"
+              if [ "$error_count" -gt 0 ]; then
+                echo "rerun=true" >> "$GITHUB_OUTPUT"
+                echo "Error message $element found in logs so the previous run has finished with that error."
+                break;
+              else
+                echo "rerun=false" >> "$GITHUB_OUTPUT"
+                echo "Error message $element not found in logs"
+              fi
+          done
```
[the remark this answers is not in the document]
Comment (andromaqui) [reply .github/workflows/rerun-failed-run.yml]: I need to get into jq its so handy - thanks for the recommendation

**Pull request merged**